corehq/apps/users/views/utils.py

```diff
@@ -115,58 +116,87 @@ def log_commcare_user_locations_changes(request, user, old_location_id, old_assi
+    """
+    Get the case count for all locations in `location_ids`
+    :returns A dict with location_id as the key and the case count as the value
+    """
+    query = (
         CaseES()
         .domain(domain)
         .owner(location_ids)
-        .user(other_commcare_user_ids)
-        .aggregation(TermsAggregation('by_location', 'owner_id')
-        .size(0))
+        .aggregation(TermsAggregation('by_location', 'owner_id').size(0))
     ).run()
-    return list(set(location_ids) - set(case_query.aggregations.by_location.keys))
-
-
-def _get_location_case_counts_with_single_user(domain, location_ids):
-    query = (CaseES()
-            .domain(domain)
-            .owner(location_ids)
-            .aggregation(TermsAggregation('by_location', 'owner_id')
-            .size(0))
-    ).run()
-    locations = SQLLocation.objects.filter(location_id__in=location_ids)
     counts = query.aggregations.by_location.counts_by_bucket()
-    return locations, counts
+    for location_id in location_ids:
+        if location_id not in counts:
+            counts[location_id] = 0
+
+    return counts
 
 
-def get_locations_with_single_user(domain, location_ids, user_id):
+def get_locations_with_orphaned_cases(domain, location_ids, user_id):
     """
     Takes a list of location IDs and returns all the given location IDs
-    where the user ID is the only one that has cases there.
+    where the user ID is the only one that has cases there. Also searched through descendant locations
+    if a location has `view_descendants` enabled.
     :param domain: The domain to search in.
     :param location_ids: A list of location IDs to get case counts on.
-    :param user_id: The user ID to check given location_ids with as the only assigned user with cases.
+    :param user_id: The user ID to check given `location_ids` with as the only assigned user with cases.
     :returns A dict with location names as the key, and the number of cases assigned to them as the value.
     """
-    # Get locations where only given user_id has cases in list of locations
-    location_ids_with_single_user = _get_location_ids_with_single_user(domain, location_ids, user_id)
-    locations_with_single_user = dict()
-    if location_ids_with_single_user:
-        locations, counts = _get_location_case_counts_with_single_user(domain, location_ids_with_single_user)
-        for location in locations:
-            if location.location_id in counts:
-                locations_with_single_user[location.name] = counts[location.location_id]
-            else:
-                locations_with_single_user[location.name] = 0
-
-    return locations_with_single_user
+    loc_ids_with_other_users = _get_location_ids_with_other_users(domain, location_ids, user_id)
+
+    # Get the assigned location ids that are in all_loc_ids but not in loc_ids_with_other_users
+    loc_ids_with_orphaned_cases = list(set(location_ids) - loc_ids_with_other_users)
+    if loc_ids_with_orphaned_cases:
```

Comment:
Should we be concerned about the situation where `loc_ids_with_orphaned_cases` is falsy but `desc_loc_ids_with_orphaned_cases` might contain some values?